frontend/src/components/LineChartComponents/LineChart.tsx

```diff
@@ -0,0 +1,760 @@
+        text: new Date(selectedDate).toLocaleDateString(i18n.language, {
+          year: 'numeric',
+          month: 'short',
+          day: '2-digit',
+        }),
+        location: 0.5,
+        background: RoundedRectangle.new(root, {
+          fill: color(theme.palette.primary.main),
+        }),
+        // Put Label to the topmost layer to make sure it is drawn on top of the axis tick labels
+        layer: Number.MAX_VALUE,
+      },
+    };
+  }, [i18n.language, root, selectedDate, theme.palette.primary.contrastText, theme.palette.primary.main]);
+  useDateAxisRange(selectedDateRangeSettings, root, chart, xAxis);
+
+  const referenceDateRangeSettings = useMemo(() => {
+    if (!referenceDay) {
+      return {};
+    }
+
+    return {
+      data: {
+        value: new Date(referenceDay).setHours(12, 0, 0),
+        above: true,
+      },
+      grid: {
+        stroke: color(darken(theme.palette.divider, 0.25)),
+        strokeOpacity: 1,
+        strokeWidth: 2,
+        strokeDasharray: [6, 4],
+      },
+    };
+  }, [referenceDay, theme.palette.divider]);
+  useDateAxisRange(referenceDateRangeSettings, root, chart, xAxis);
+
+  const setReferenceDayX = useCallback(() => {
+    if (!chart || !root || !xAxis || !referenceDay) {
+      return;
+    }
+
+    const midday = new Date(referenceDay).setHours(12, 0, 0);
+
+    const xAxisPosition = xAxis.width() * xAxis.toGlobalPosition(xAxis.dateToPosition(new Date(midday)));
+    const globalPosition = xAxis.toGlobal({x: xAxisPosition, y: 0});
+    const docPosition = root.rootPointToDocument(globalPosition).x;
+    setReferenceDayBottom(docPosition);
+  }, [chart, root, xAxis, referenceDay, setReferenceDayBottom]);
+
+  // Effect to update reference day position
+  useLayoutEffect(() => {
+    if (!root || !chart || !xAxis) {
+      return;
+    }
+
+    setReferenceDayX();
+    const minEvent = xAxis.onPrivate('selectionMin', setReferenceDayX);
+    const maxEvent = xAxis.onPrivate('selectionMax', setReferenceDayX);
+    const seriesEvent = chart.series.events.on('push', (ev) => {
+      ev.newValue.events.on('boundschanged', setReferenceDayX);
+    });
+
+    const resizeObserver = new ResizeObserver(setReferenceDayX);
+    resizeObserver.observe(root.dom);
+
+    return () => {
+      resizeObserver.disconnect();
+      minEvent.dispose();
+      maxEvent.dispose();
+      seriesEvent.dispose();
+    };
+    // This effect should re-run when root, chart and xAxis are initialized
+  }, [root, chart, xAxis, setReferenceDayX]);
+
+  const lineChartDataSettings = useMemo(() => {
+    if (!root || !xAxis || !yAxis || !lineChartData) {
+      return [];
+    }
+
+    return lineChartData.map((line) => ({
+      xAxis: xAxis,
+      yAxis: yAxis,
+      id: `${chartId}_${line.serieId}`,
+      name: line.name
+        ? localization.overrides && localization.overrides[line.name]
+          ? customT(localization.overrides[line.name])
+          : defaultT(line.name)
+        : '',
```

Comment:
This is really hard to read. Maybe compute the name before the return in a more readable form.